OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessage.java

```diff
@@ -76,11 +82,31 @@
         this.variants = parseVariants(json.getJSONObject(IAM_VARIANTS));
         this.triggers = parseTriggerJson(json.getJSONArray(IAM_TRIGGERS));
         this.clickedClickIds = new HashSet<>();
+        this.endTime = parseEndTimeJson(json);
 
         if (json.has(IAM_REDISPLAY_STATS))
             this.redisplayStats = new OSInAppMessageRedisplayStats(json.getJSONObject(IAM_REDISPLAY_STATS));
     }
 
+    private Date parseEndTimeJson(JSONObject json) {
+        String endTimeString;
+        try {
+            endTimeString = json.getString(END_TIME);
+        } catch (JSONException e) {
+            return null;
+        }
+
+        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US);
```

Comment:
can we put this on a constant?

Reply:
Can we have the same iOS object https://github.com/OneSignal/OneSignal-iOS-SDK/pull/841#discussion_r547501668 in here?

Reply:
created a OneSignalSimpleDateFormatter class with a static method to get the iso8601 format